cmd/nerdctl/container_create.go

```diff
@@ -50,36 +52,529 @@ func newCreateCommand() *cobra.Command {
+		return
+	}
+	sysctl, err := cmd.Flags().GetStringArray("sysctl")
+	if err != nil {
+		return
+	}
+	// #endregion
+
+	// #region for volume flags
+	volume, err := cmd.Flags().GetStringArray("volume")
+	if err != nil {
+		return
+	}
+	// tmpfs needs to be StringArray, not StringSlice, to prevent "/foo:size=64m,exec" from being split to {"/foo:size=64m", "exec"}
+	tmpfsSlice, err := cmd.Flags().GetStringArray("tmpfs")
+	if err != nil {
+		return
+	}
+	mountsSlice, err := cmd.Flags().GetStringArray("mount")
+	if err != nil {
+		return
+	}
+	// #endregion
+
+	// #region for rootfs flags
+	readOnly, err := cmd.Flags().GetBool("read-only")
+	if err != nil {
+		return
+	}
+	imageless, err := cmd.Flags().GetBool("rootfs")
+	if err != nil {
+		return
+	}
+	// #endregion
+
+	// #region for env flags
+	entrypoint, err := cmd.Flags().GetStringArray("entrypoint")
+	if err != nil {
+		return
+	}
+	workdir, err := cmd.Flags().GetString("workdir")
+	if err != nil {
+		return
+	}
+	env, err := cmd.Flags().GetStringArray("env")
+	if err != nil {
+		return
+	}
+	envFile, err := cmd.Flags().GetStringSlice("env-file")
+	if err != nil {
+		return
+	}
+	// #endregion
+
+	// #region for metadata flags
+	nameChanged := cmd.Flags().Changed("name")
+	name, err := cmd.Flags().GetString("name")
+	if err != nil {
+		return
+	}
+	label, err := cmd.Flags().GetStringArray("label")
+	if err != nil {
+		return
+	}
+	labelsFilePath, err := cmd.Flags().GetStringSlice("label-file")
+	if err != nil {
+		return
+	}
+	labelsFilePath = strutil.DedupeStrSlice(labelsFilePath)
+	cidfile, err := cmd.Flags().GetString("cidfile")
+	if err != nil {
+		return
+	}
+	var pidFile string
+	if cmd.Flags().Lookup("pidfile").Changed {
+		pidFile, err = cmd.Flags().GetString("pidfile")
+		if err != nil {
+			return
+		}
+	}
+	// #endregion
+
+	// #region for logging flags
+	// json-file is the built-in and default log driver for nerdctl
+	logDriver, err := cmd.Flags().GetString("log-driver")
+	if err != nil {
+		return
+	}
+	logOptArray, err := cmd.Flags().GetStringArray("log-opt")
+	if err != nil {
+		return
+	}
+	// #endregion
+
+	// #region for shared memory flags
+	ipc, err := cmd.Flags().GetString("ipc")
+	if err != nil {
+		return
+	}
+	shmSize, err := cmd.Flags().GetString("shm-size")
+	if err != nil {
+		return
+	}
+	// #endregion
+
+	// #region for gpu flags
+	gpus, err := cmd.Flags().GetStringArray("gpus")
+	if err != nil {
+		return
+	}
+	// #endregion
+
+	// #region for ulimit flags
+	ulimits, err := cmd.Flags().GetStringSlice("ulimit")
+	if err != nil {
+		return
+	}
+	// #endregion
+
+	// #region for verify flags
+	verifier, err := cmd.Flags().GetString("verify")
+	if err != nil {
+		return
+	}
+	cosignKey, err := cmd.Flags().GetString("cosign-key")
+	if err != nil {
+		return
+	}
+	// #endregion
+	// #region for ipfs flags
+	ipfsAddressStr, err := cmd.Flags().GetString("ipfs-address")
+	if err != nil {
+		return
+	}
+	// #endregion
+
+	return types.ContainerCreateOptions{
```

Comment:
nit: you can assign the value from `cmd` directly to corresponding field in `opt`, thus avoiding the whole repeated assignment again in line 457-548. For example:

```golang
	opt.IPFSAddress, err = cmd.Flags().GetString("ipfs-address")
	if err != nil {
		return
	}
```

I know writing this is tedious so no change is requested here :) I'm okay with this format. Actually, if you plan to make the change, I'd suggest to make a copy of the current format just in case other reviewers prefer this (more consistent with other `processXXXOptions`, though verbose)

Reply:
SGTM. 
I didn't notice that this consistent style has changed, and I will change it.